src/components/NewsletterSignup/NewsletterSignup.tsx

```diff
@@ -0,0 +1,238 @@
+  /** ID that other components can cross-reference for accessibility purposes */
+  id?: string;
+  /** Toggles the invalid state for the email field. */
+  isInvalidEmail?: boolean;
+  /** Value to determine the section color highlight */
+  newsletterSignupType?: SectionTypes;
+  /** A handler function that will be called when the form is submitted. */
+  onSubmit: (event: React.FormEvent<any>) => void;
+  /** A handler function that will be called when the text input changes. */
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  /** Link to the relevant privacy policy page. */
+  privacyPolicyLink?: string;
+  /** Used to populate the `<h3>` header title. */
+  title?: string;
+  /** The value of the email text input field. */
+  valueEmail?: string;
+  /** Used to specify what is displayed in the component form/feedback area. */
+  view?: NewsletterSignupViewType;
+}
+
+export type NewsletterSignupViewType =
+  | "form"
+  | "submitting"
+  | "confirmation"
+  | "error";
+
+const defaultDescriptionText =
+  "Stay connected with the latest research news from NYPL, including information about our events, programs, " +
+  "exhibitions, and collections.";
+
+/**
+ * The NewsletterSignup component provides a way for patrons to register for an
+ * email-based newsletter distribution list.
+ */
+export const NewsletterSignup = chakra(
+  forwardRef<HTMLDivElement, NewsletterSignupProps>(
+    (
+      {
+        className,
+        confirmationHeading,
+        confirmationText,
+        descriptionText = defaultDescriptionText,
+        errorHeading = "Oops! Something went wrong.",
+        errorText,
+        formHelperText,
+        id,
+        isInvalidEmail = false,
+        newsletterSignupType = "whatsOn",
+        onChange,
+        onSubmit,
+        privacyPolicyLink = "https://www.nypl.org/help/about-nypl/legal-notices/privacy-policy",
+        valueEmail,
+        title = "Sign Up for Our Newsletter!",
+        view = "form",
+        ...rest
+      },
+      ref?
+    ) => {
+      const { isLargerThanMobile } = useNYPLBreakpoints();
+      const styles = useMultiStyleConfig("NewsletterSignup", {
+        newsletterSignupType,
+      });
+      const iconColor = useColorModeValue(null, "dark.ui.typography.body");
+
+      const isFormView = view === "form" || view === "submitting";
+
+      // Manage focus to ensure accessibility when confirmation or error message is rendered.
+      const focusRef = React.useRef<HTMLDivElement>(null);
+      // When view prop changes, set focus to the confirmation or error content element.
+      React.useEffect(() => {
+        focusRef.current?.focus();
+      }, [view]);
+      return (
+        <Stack
+          direction={isLargerThanMobile ? "row" : "column"}
+          ref={ref}
+          __css={styles}
+          {...rest}
+        >
+          <VStack __css={styles.pitch} alignItems="flex-start">
+            {title && <Heading level="h3" text={title} margin="unset" />}
```

Comment:
That you mention this, I think the entire `title` prop _should_ be dictated by the consuming app. The DS has an expectation of how and where this component should go on a page, but if we set this to h3 or h2, we are now telling the app how to structure their app around this component. I don't think we should have that much power. If this becomes a small part on the footer of the page that doesn't follow the app's heading hierarchy, we're going to cause accessibility issues.

So perhaps we should:

- make `title: JSX.Element` as a prop
- set this line as `{title}`
- use this as `<NewsletterSignup title={<Heading level="h3" text={title} margin="unset" />} ... />`

Then apps will have more control.

Reply:
As Edwin mentioned I started testing the component integration with Scout. I did  a quick inspection of the heading:
- Our heading prop was type string 
BUT
- It renders as a `h2` with `font-size=28px`, with the NewsletterSignup component test it renders a `h3` with `font-size=32px`.
I would agree with Edwin that it would make sense to allow a customized heading.

Reply:
Yes, go ahead and update the title prop to accept the `Heading` component instead of a plain string.  Make sure to also update the component docs accordingly.  I will update the spec doc to reflect this change.

Reply:
But let's also change the default to render as an `<h2>` with `size="heading3"`.

Reply:
I added this and added a line to the accessibility text too. Tests are updated and it's the args table too. I did remove the controls as it seems that storybook handle it.